### PR TITLE
po: Put translations into individual packages

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by petervo** [comment]
Is there a reason not to use ```msggrep``` for spliting the po file data?

**Comment by mvollmer** [comment]
> Is there a reason not to use `msggrep` for spliting the po file data?

After thinking more about this, no.

I have totally over-engineered this, I am afraid.  Even after understanding that the splitting is not done for  the benefit of 3rd party packages, I tried to create a mechanism that could be eventually generalized for out-of-tree packages.  But as you also said, 3rd party packages are unlikely to be interested in gettext...

So, at least I caught up with how we use webpack... :-)

**Comment by mvollmer** [comment]
I think we can keep `manifest2po`, though.

**Comment by mvollmer** [comment]
The override translations in pkg/playground need work.

**Comment by mvollmer** [comment]
> The override translations in pkg/playground need work.

Done.  Simply renaming them to `extra.de.po` is enough.

File: tools/webpack-make [hunk 1]
```diff
@@ -149,13 +151,28 @@ function generateDeps(makefile, stats) {
...
+
     makeArray(prefix + "_INSTALL", installs);
     makeArray(prefix + "_DEBUG", debugs);
     makeArray(prefix + "_TESTS", tests);
 
+    var filters = "$(addprefix -N ,$(" + prefix + "_INPUTS))";
+    if (prefix === "shell")
+        filters += " $(addprefix -N ,$(shell find pkg/ -name manifest.json))"
+
+    lines.push(path.join(pkgdir, "/%.po") + ": po/%.po");
```
Comment (mvollmer) [inline tools/webpack-make]: "/%.po" should have no "/".

**Comment by mvollmer** [comment]
The po files don't seem to make it into the packages.

**Comment by mvollmer** [comment]
> The po files don't seem to make it into the packages.

I misunderstood the significance of the empty `po.js` files.  They were missing.

File: tools/webpack-make (hunk 1 above)
Comment (mvollmer) [inline tools/webpack-make]: Fixed

**Comment by mvollmer** [comment]
When making the sources in tmp-dist, the po files stay empty because the msggrep filter uses wrong filenames.

**Comment by petervo** [comment]
@mvollmer did you pull and push to/from zanata when updating the po files? If not probably better to leave them out of these commits and I can add another one with that.

**Comment by mvollmer** [comment]
> did you pull and push to/from zanata when updating the po files?

No, but I ran "make update-po" as part of "po: Accumulate all locations in cockpit.pot" and "shell: Include manifest strings in translations" because those commit change the locations in cockpit.pot and the po files.

> If not probably better to leave them out of these commits and I can add another one with that.

Okay.

**Comment by mvollmer** [comment]
Needs unit tests for the resource globbing.

**Comment by mvollmer** [comment]
> Needs unit tests for the resource globbing.

Added one.

File: src/bridge/test-packages.c [hunk 2]
```diff
@@ -526,6 +528,37 @@ test_list_bad_name (TestCase *tc,
...
+};
+
+static void
+test_glob (TestCase *tc,
+           gconstpointer fixture)
+{
+  GError *error = NULL;
+  GBytes *message;
+  JsonObject *object;
+  guint count;
```
Comment (petervo) [inline src/bridge/test-packages.c]: Used, making tests fail.

Comment (mvollmer) [inline src/bridge/test-packages.c]: Removed.

**Pull request closed without merging**